### PR TITLE
[RISCV] Implement XCValu Intrinsics

### PR DESCRIPTION
**Pull request by realqhc**
Implement XcvAlu LLVM intrinsics.

Spec: https://github.com/openhwgroup/cv32e40p/blob/62bec66b36182215e18c9cf10f723567e23878e9/docs/source/instruction_set_extensions.rst

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVExpandPseudoInsts.cpp [hunk 1]
```diff
@@ -395,6 +409,87 @@ bool RISCVExpandPseudo::expandCoreVBitManip(MachineBasicBlock &MBB,
...
+      break;
+    case RISCV::CV_CLIP_PSEUDO:
+      Opcode = RISCV::CV_CLIP;
+      break;
+    default:
+      llvm_unreachable("unknown instruction");
+  }
+  const MCInstrDesc &Desc = TII->get(Opcode);
+  BuildMI(MBB, MBBI, DL, Desc, DstReg).
+      addReg(I).
```
Comment (melonedo) [inline llvm/lib/Target/RISCV/RISCVExpandPseudoInsts.cpp]: Probably `.addReg(I)` instead of `addReg(I).`?

**Comment by lewis-revill** [comment]
Looks good, should remember when we come to implement the Clang side of this that it would be nice to have Clang do verification on `clip`/`clipu` to check that the argument is valid (power of two minus 1). That way we would get a nice frontend error rather than a backend error.

File: llvm/lib/Target/RISCV/RISCVExpandPseudoInsts.cpp (hunk 1 above)
Comment (lewis-revill) [reply llvm/lib/Target/RISCV/RISCVExpandPseudoInsts.cpp]: Agreed

**Comment by simonpcook** [comment]
Could someone with write access merge this, I think a bug should be opened with Lewis' comment afterwards as a reminder for however this is used in Clang (possibly part of #74 discussion?)

**Pull request merged**